# Copyright 2007 Sun Microsystems, Inc.  All rights reserved.
# Use subject to license terms.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the
# "Software"), to deal in the Software without restriction, including
# without limitation the rights to use, copy, modify, merge, publish,
# distribute, and/or sell copies of the Software, and to permit persons
# to whom the Software is furnished to do so, provided that the above
# copyright notice(s) and this permission notice appear in all copies of
# the Software and that both the above copyright notice(s) and this
# permission notice appear in supporting documentation.
# 
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
# OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
# HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
# INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
# FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
# NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
# WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
# 
# Except as contained in this notice, the name of a copyright holder
# shall not be used in advertising or otherwise to promote the sale, use
# or other dealings in this Software without prior written authorization
# of the copyright holder.

Make docbook2txt work on Solaris for Xorg doc builds

diff -urp -x '*~' -x '*.orig' backends/txt backends/txt
--- backends/txt	2004-02-11 05:58:03.000000000 -0800
+++ backends/txt	2007-01-29 21:07:20.359122000 -0800
@@ -1,3 +1,4 @@
+#! /usr/bin/bash
 # Backend to convert something into ASCII text
 # Send any comments to Eric Bischoff <eric@caldera.de>
 # This program is under GPL license. See LICENSE file for details.
@@ -6,6 +7,10 @@ if [ -x /usr/bin/lynx ]
 then
   CONVERT=/usr/bin/lynx
   ARGS="-force_html -dump -nolist -width=72"
+elif [ -x /opt/sfw/bin/lynx ]
+then
+  CONVERT=/opt/sfw/bin/lynx
+  ARGS="-force_html -dump -nolist -width=72"
 elif [ -x /usr/bin/links ]
 then
   CONVERT=/usr/bin/links
diff -urp -x '*~' -x '*.orig' bin/jw.in bin/jw.in
--- bin/jw.in	2003-04-30 09:21:49.000000000 -0700
+++ bin/jw.in	2007-01-29 21:07:20.359852000 -0800
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 # Jade Wrapper
 # Script to convert a SGML file into some other format
 # Send any comments to Eric Bischoff <eric@caldera.de>
@@ -77,7 +77,8 @@ SGML_EXTRA_CATALOGS=""
 
 # Set SGML base directory and centralized catalogs directory
 SGML_BASE_DIR="@prefix@/share/sgml"
-SGML_CATALOGS_DIR="/etc/sgml"
+SGML_CATALOGS_DIR="/usr/share/sgml"
+SGML_SEARCH_PATH=$SGML_SEARCH_PATH:/usr/share/lib/sgml/locale/C/entities:/usr/share/sgml/db42xml ; export SGML_SEARCH_PATH
 if [ -f "$SGML_CONF" ]
 then
   RE='^[:space:]*SGML_BASE_DIR[:space:]*=[:space:]*'
@@ -290,7 +291,9 @@ fi
 
 # Use SGML declaration for XML if needed
 if [ "$SGML_DECL" = "" -a "$SGML_XML" = "xml" ]
-then SGML_DECL=${SGML_BASE_DIR}/xml.dcl
+then SGML_DECL=/usr/share/sgml/openjade/xml.dcl
+elif [ "$SGML_DECL" = "" ]
+then SGML_DECL=/usr/share/sgml/docbook_4.2/docbook.dcl
 fi
 
 # Try to find the SGML normalizer
@@ -306,13 +309,13 @@ fi
 # Set path to SGML catalogs (first try centralized catalog)
 case $SGML_STANDARD_CATALOGS in
   yes)	export SGML_CATALOGS_DIR SGML_FILE SGML_XML SGML_NORM
-	SGML_CENTRALIZED_CATALOG=`sh $SGML_FRONTEND centralized-catalog`
+	SGML_CENTRALIZED_CATALOG=`bash $SGML_FRONTEND centralized-catalog`
 	if [ -s $SGML_CENTRALIZED_CATALOG ]
 	then
 	  SGML_CATALOG_FILES=$SGML_CENTRALIZED_CATALOG
 	else
-	  SGML_CATALOG_FILES=`find $SGML_BASE_DIR -name catalog`
-	  SGML_CATALOG_FILES=`echo "$SGML_CATALOG_FILES" | tr [:space:] :`
+	  SGML_CATALOG_FILES=`find $SGML_BASE_DIR /usr/share/sgml -name catalog`
+	  SGML_CATALOG_FILES=`echo "$SGML_CATALOG_FILES" | tr '[:space:]' :::::::`
 	fi
 	;;
   no)	SGML_CATALOG_FILES=""
@@ -340,7 +343,7 @@ esac
 if [ "$SGML_STYLESHEET" = "default" -o "$SGML_STYLESHEET" = "none" ]
 then
   export SGML_BASE_DIR SGML_TYPE SGML_STYLESHEET
-  SGML_STYLESHEET=`sh $SGML_FRONTEND style-sheet`
+  SGML_STYLESHEET=`bash $SGML_FRONTEND style-sheet`
   SGML_RETURN=$?
   if [ $SGML_RETURN -gt 0 ]
   then exit `expr 7 + $SGML_RETURN`
@@ -428,9 +431,9 @@ export SGML_CATALOG_FILES SGML_BASE_DIR 
 NOCHUNKS=`echo $SGML_OPTIONS | grep nochunks`
 if [ -z "$NOCHUNKS" ]
 then
-	sh $SGML_BACKEND
+	bash $SGML_BACKEND
 else
-	sh $SGML_BACKEND >$SGML_FILE_NAME.html
+	bash $SGML_BACKEND >$SGML_FILE_NAME.html
 fi
 SGML_RETURN=$?
 cd $SGML_CURRENT_DIRECTORY
diff -urp -x '*~' -x '*.orig' doc/HTML/Makefile.am doc/HTML/Makefile.am
--- doc/HTML/Makefile.am	2002-04-22 04:48:20.000000000 -0700
+++ doc/HTML/Makefile.am	2007-01-29 21:07:20.360396000 -0800
@@ -23,8 +23,8 @@ $(html_DATA): $(top_srcdir)/doc/docbook-
 		$(top_srcdir)/doc/refentry/docbook2texi-spec.pl.sgml \
 		$(top_srcdir)/doc/refentry/frontend-spec.sgml \
 		$(top_srcdir)/doc/refentry/sgmldiff.sgml
-	SGML_CATALOG_FILES=/etc/sgml/catalog \
-	SGML_SEARCH_PATH=$(top_srcdir):$(top_srcdir)/doc:.. \
+	SGML_CATALOG_FILES=/etc/sgml/catalog:/usr/share/sgml/docbook_3/docbook.cat:/usr/share/sgml/docbook/dsssl-stylesheets/catalog:/usr/share/sgml/openjade/catalog \
+	SGML_SEARCH_PATH=$(top_srcdir):$(top_srcdir)/doc:..:/usr/share/lib/sgml/locale/C/entities \
 		jade -t sgml -i html -d $(top_srcdir)/docbook-utils.dsl\#html \
 			-V '%use-id-as-filename%' $<
 
diff -urp -x '*~' -x '*.orig' frontends/docbook.in frontends/docbook.in
--- frontends/docbook.in	2002-11-02 09:45:10.000000000 -0800
+++ frontends/docbook.in	2007-01-29 21:07:20.360680000 -0800
@@ -15,7 +15,7 @@ case "$1" in
 	SGML_VERSION=`"$SGML_NORM" $SGML_NORM_OPTS -- "$SGML_FILE" 2>/dev/null |
 			grep -i '<!DOCTYPE' |
 			head -n 1 |
-			sed 's/^.*DocBook\( XML\)\? V\([0-9][\.0-9]*\).*$/\2/'`
+			perl -p -e 's/^.*DocBook\( XML\)\? V\([0-9][\.0-9]*\).*$/\2/'`
 	SGML_CATALOG=
 	for f in "${SGML_CATALOGS_DIR}/${SGML_XML}-docbook-${SGML_VERSION}"*".cat"
 	do 
@@ -41,7 +41,7 @@ case "$1" in
 	    echo "${SGML_BASE_DIR}/docbook/utils-@VERSION@/docbook-utils.dsl#${SGML_TYPE}"
 	    ;;
 	  none)
-	    find $SGML_BASE_DIR -name docbook.dsl | grep "$SGML_TYPE/docbook.dsl" | awk '{print $1}'
+	    find /usr/share/sgml -name docbook.dsl | grep "$SGML_TYPE/docbook.dsl" | awk '{print $1}'
 	    ;;
 	esac
         ;;
